components/ProfileDropdownMenu.tsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react'
+import Dropdown from 'react-bootstrap/Dropdown'
+import UserInfoImage from './UserInfoImage'
+import { ChevronDownIcon } from '@primer/octicons-react'
+import LogoutContainer from './LogoutContainer'
+import { Button } from './theme/Button'
+import { UserInfo } from '../@types/user'
+import { useUserInfoQuery } from '../graphql/index'
+import { useRouter } from 'next/router'
+import _ from 'lodash'
+import styles from '../scss/profileDropDown.module.scss'
+
+type ProfileDropDownMenuProps = {
+  username: string
+  isAdmin: boolean
+}
+
+const ProfileDropdownMenu: React.FC<ProfileDropDownMenuProps> = ({
+  username,
+  isAdmin
+}) => {
+  const { data } = useUserInfoQuery({
+    variables: { username },
+    skip: !username
+  })
+  const router = useRouter()
+  const location = router.asPath
+  const PROFILE_PATH = '/profile/' + username
+
+  const fullname = _.get(data, 'userInfo.user.name', '')
+
+  const userInfo: UserInfo = {
+    // 'A' stands for Anonymous, in case user did not put in full name
+    username,
+    firstName: fullname.split(' ')[0] || 'A',
+    lastName: fullname.split(' ')[1] || ' ',
+    discordUserId: _.get(data, 'userInfo.user.discordUserId', ''),
+    discordUsername: _.get(data, 'userInfo.user.discordUsername', ''),
+    discordAvatarUrl: _.get(data, 'userInfo.user.discordAvatarUrl', '')
+  }
+
+  const dropdownAdminMenuItems = [
+    { title: 'Lessons', path: '/admin/lessons' },
+    { title: 'Users', path: '/admin/users' },
+    { title: 'Alerts', path: '/admin/alerts' }
+  ]
+
+  const isActive = (path: string) => {
+    if (path === location) return `${styles['active']}`
+    return ''
+  }
+
+  const adminDropdownMenu = dropdownAdminMenuItems.map(({ title, path }) => (
+    <Dropdown.Item
+      className={`${styles['dropdown-item']}`}
+      href={path}
+      key={title}
+      bsPrefix={isActive(path)}
+    >
+      {title}
+    </Dropdown.Item>
+  ))
+
+  const buttonMenuToggle = React.forwardRef(
+    (props, ref: React.Ref<HTMLDivElement>) => (
```

Comment:
Why does ref have a type definition, but props doesn't?

Reply:
Actually we have no unique fields in the props object so this is ok

Reply:
I think this might throw typescript error saying children doesn't exist in props

Reply:
> by passing it generic types

What do you mean?

Reply:
i am not familiar yet with all the type stuff so I used `<HTMLDivElement>` as a type for forwardRef and tsc didn't scream.

Reply:
This guy had code similar to yours and it ran into issue of children not appearing in props object
https://stackoverflow.com/questions/54654303/using-a-forwardref-component-with-children-in-typescript

In the documentation, it looks like we have to use `props.children` whenever the dropdown component we use contains children
https://react-bootstrap.github.io/components/dropdowns/

I tried it out on a typescript playground too and it gives me the error of children not existing in props
https://codesandbox.io/s/forwardref-typescript-c0b4r?file=/src/App.tsx
Try this code in the `App.tsx` file:
```
import * as React from "react";


const Search = React.forwardRef<HTMLDivElement>((props, ref) => {
  return <div ref={ref} {...props}>{props.children}</div>;
});
Search.displayName = "Search";

export default function App() {
  const input = React.useRef<HTMLInputElement>(null);
  React.useEffect(() => {
    if (input.current) {
      input.current.focus();
    }
  }, []);
  return <Search ref={input}>Banana</Search>;
}
```

I am not sure why the error does it appear on your end, but either way since we are using typescript, we should make sure to have a type definition for all parameters.

Reply:
After reading this: https://stackoverflow.com/a/66876771
It is suggested to either create a type with a children prop, use `React.PropsWithChildren` or use `React.ComponentPropsWithoutRef` if we want to forward all the props, i used the later so that it passes all the props.